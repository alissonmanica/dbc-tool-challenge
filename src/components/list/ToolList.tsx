import { useDispatch, useSelector } from 'react-redux'
import { ContainerList } from './ToolList.styles'
import toolSelector from '../../store/tools/selector'
import ToolListItem from './ToolListItem'
import AddTool from '../modal/AddTool'
import { useEffect } from 'react'
import { toolActions } from '../../store/tools/duck'

function ToolList() {
  const tools = useSelector(toolSelector.get)
  const openModalAdd = useSelector(toolSelector.openModalAdd)

  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(toolActions.loadTools())
  }, [])

  return (
    <>
      {openModalAdd && <AddTool />}
      <ContainerList>
        {tools?.map((itemTools, index) => (
          <ToolListItem key={index} tool={itemTools} />
        ))}
      </ContainerList>
    </>
  )
}
export default ToolList
