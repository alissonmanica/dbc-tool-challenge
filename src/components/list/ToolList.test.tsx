import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event/dist/types/setup'
import ToolList from './ToolList'

describe('should test the behavior of the list', () => {
  it('should start the list without size', () => {
    const number = 5
    expect(number).toBe(5)
    // screen.debug(list as any)
  })
})
