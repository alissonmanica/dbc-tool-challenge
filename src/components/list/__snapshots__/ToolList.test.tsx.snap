// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should test snapshot for ToolList component 1`] = `
<div
  className="sc-bczRLJ knepsb"
/>
`;
